src/core/table/table.jsx

```diff
@@ -0,0 +1,268 @@
+            newSelected = newSelected.concat(selected, rawInfo);
+        } else if (selectedIndex === 0) {
+            newSelected = newSelected.concat(selected.slice(1));
+        } else if (selectedIndex === selected.length - 1) {
+            newSelected = newSelected.concat(selected.slice(0, -1));
+        } else if (selectedIndex > 0) {
+            newSelected = newSelected.concat(
+                selected.slice(0, selectedIndex),
+                selected.slice(selectedIndex + 1)
+            );
+        }
+
+        if (!multiSelect) {
+            newSelected = [rawInfo];
+        }
+
+        setSelected(newSelected);
+        onRowSelect(newSelected);
+        onRowClick(rawInfo);
+    };
+
+    const createSortHandler = (property) => (event) => {
+        handleRequestSort(event, property);
+    };
+
+    const isSelected = (id) => selected.findIndex((entry) => entry._id === id) !== -1;
+
+    const emptyRows = page > 0 ? Math.max(0, (1 + page) * rowsPerPage - processedData.length) : 0;
+
+    const renderTableHead = () => {
+        return (
+            <TableHead>
+                <TableRow>
+                    <TableCell></TableCell>
+                    {processedColumns.map(({ id, accessor = '', label, style }) => {
+                        return (
+                            <TableCell
+                                sx={{ ...style }}
+                                key={id}
+                                sortDirection={orderBy === accessor ? order : false}>
+                                <TableSortLabel
+                                    active={orderBy === accessor}
+                                    direction={orderBy === accessor ? order : 'asc'}
+                                    onClick={createSortHandler(accessor)}
+                                >
+                                    {label}
+                                    {orderBy === accessor ? (
+                                        <Box component="span" sx={visuallyHidden}>
+                                            {order === 'desc' ? 'sorted descending' : 'sorted ascending'}
+                                        </Box>
+                                    ) : null}
+                                </TableSortLabel>
+                            </TableCell>
+                        )
+                    })}
+                </TableRow>
+            </TableHead>
+        )
+    }
+
+    const handleChangePage = (event, newPage) => {
+        setPage(newPage);
+    };
+
+    const handleChangeRowsPerPage = (event) => {
+        setRowsPerPage(parseInt(event.target.value, 10));
+        setPage(0);
+    };
+
+    function descendingComparator(a, b, orderBy) {
+        if (b[orderBy] < a[orderBy]) {
+            return -1;
+        }
+        if (b[orderBy] > a[orderBy]) {
+            return 1;
+        }
+        return 0;
+    }
+
+    function getComparator(order, orderBy) {
+        return order === "desc"
```

Comment:
I see there is a code repetition, it could be shortened to:

```suggestion
        const scalar = order === "desc" ? 1 : -1;
        return scalar * descendingComparator(a.original, b.original, orderBy);
```